ui-v2/app/templates/dc/nodes/index.hbs

```diff
@@ -1,137 +1,106 @@
-                    <span data-test-leader={{leader.Address}} data-tooltip="Leader">Leader</span>
-                  {{/if}}
-                </BlockSlot>
-              </HealthcheckedResource>
-            </GridCollection>
-          </BlockSlot>
-          <BlockSlot @name="empty">
-            <EmptyState>
-              <BlockSlot @name="header">
-                <h2>No nodes found</h2>
-              </BlockSlot>
-              <BlockSlot @name="body">
-                <p>
-                  There don't seem to be any nodes matching that search.
-                </p>
-              </BlockSlot>
-            </EmptyState>
-          </BlockSlot>
-        </ChangeableSet>
-      </div>
-  {{/if}}
-{{/let}}
-{{#if (eq items.length 0) }}
-      <EmptyState @allowLogin={{true}}>
-        <BlockSlot @name="header">
-          <h2>Welcome to Nodes</h2>
+        class="with-sort"
+      >
+        <BlockSlot @name="secondary">
+          <PopoverSelect
+            @position="right"
+            @onchange={{action (mut sortBy) value='target.selected'}}
+            @multiple={{false}}
+          as |components|>
+            <BlockSlot @name="selected">
+              <span>
+                {{#let (from-entries (array
+                      (array "Node:asc" "A to Z")
+                      (array "Node:desc" "Z to A")
+                      (array "Status:asc" "Unhealthy to Healthy")
+                      (array "Status:desc" "Healthy to Unhealthy")
+                    ))
+                  as |selectable|
+                }}
+                  {{get selectable sort}}
+                {{/let}}
+              </span>
+            </BlockSlot>
+            <BlockSlot @name="options">
+  {{#let components.Optgroup components.Option as |Optgroup Option|}}
+              <Optgroup @label="Name">
+                <Option @value="Node:asc" @selected={{eq "Node:asc" sort}}>A to Z</Option>
+                <Option @value="Node:desc" @selected={{eq "Node:desc" sort}}>Z to A</Option>
+              </Optgroup>
+              <Optgroup @label="Health Status">
+                <Option @value="Status:asc" @selected={{eq "Status:asc" sort}}>Unhealthy to Healthy</Option>
+                <Option @value="Status:desc" @selected={{eq "Status:desc" sort}}>Healthy to Unhealthy</Option>
+              </Optgroup>
+  {{/let}}
+            </BlockSlot>
+          </PopoverSelect>
         </BlockSlot>
-        <BlockSlot @name="body">
-          <p>
-            There don't seem to be any nodes, or you may not have access to view nodes yet.
-          </p>
+      </SearchBar>
+    {{/if}}
+    </BlockSlot>
+    <BlockSlot @name="content">
+    {{#let (sort-by (comparator 'node' sort) items) as |sorted|}}
+      <ChangeableSet @dispatcher={{searchable 'node' sorted}} @terms={{search}}>
+        <BlockSlot @name="set" as |filtered|>
+          <ConsulNodeList @items={{filtered}} @leader={{leader}}>
+            <BlockSlot @name="empty">
```

Comment:
The new `ConsulNodeList` component doesn't have any slots in it and it doesn't yield anything, so this can just be an inline `<ConsulNodeList @items={{filtered}} @leader={{leader}} />` I think?

Actually looking at this again I think you need an `empty` slot on the `ChangeableSet` rather than the `ConsulNodeList`. I think maybe the `services/index.hbs` page is the best one to use as a guide here.

Reply:
Moving the EmptyState outside of ConsulNodeList, but within ChangeableSet leaves a 'Login with a specified token` button. I'll just leave the EmptySate outside.